apps/crn-server/test/data-providers/contentful/working-groups.data-provider.test.ts

```diff
@@ -0,0 +1,651 @@
+          {
+            id: 'hub@asap.science',
+            name: 'ASAP Hub',
+            color: '#B1365F',
+            groups: [],
+            workingGroups: [],
+          },
+        ]);
+      });
+
+      test('should return empty calendars if there is not any', async () => {
+        const id = 'some-id';
+        const contentfulGraphQLResponse =
+          getContentfulWorkingGroupGraphqlResponse();
+        contentfulGraphQLResponse.workingGroups!.calendars = null;
+        contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+          contentfulGraphQLResponse,
+        );
+        const response = await workingGroupDataProvider.fetchById(id);
+
+        expect(response?.calendars).toEqual([]);
+      });
+    });
+
+    describe('deliverables', () => {
+      test('should return deliverables if there is any', async () => {
+        const id = 'some-id';
+        const contentfulGraphQLResponse =
+          getContentfulWorkingGroupGraphqlResponse();
+        contentfulGraphQLResponse.workingGroups!.deliverablesCollection = {
+          items: [
+            {
+              description: 'Deliverable 1',
+              status: 'In Progress',
+            },
+            {
+              description: 'Deliverable 2',
+              status: 'Complete',
+            },
+          ],
+        };
+        contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+          contentfulGraphQLResponse,
+        );
+        const response = await workingGroupDataProvider.fetchById(id);
+
+        expect(response?.deliverables).toEqual([
+          { description: 'Deliverable 1', status: 'In Progress' },
+          { description: 'Deliverable 2', status: 'Complete' },
+        ]);
+      });
+
+      test('should return empty deliverables if they do not exist', async () => {
+        const id = 'some-id';
+        const contentfulGraphQLResponse =
+          getContentfulWorkingGroupGraphqlResponse();
+        contentfulGraphQLResponse.workingGroups!.deliverablesCollection = null;
+        contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+          contentfulGraphQLResponse,
+        );
+        const response = await workingGroupDataProvider.fetchById(id);
+
+        expect(response?.deliverables).toEqual([]);
+      });
+
+      test('should filter null deliverablesCollection items', async () => {
+        const id = 'some-id';
+        const contentfulGraphQLResponse =
+          getContentfulWorkingGroupGraphqlResponse();
+        contentfulGraphQLResponse.workingGroups!.deliverablesCollection = {
+          items: [
+            null,
+            {
+              description: 'Deliverable 1',
+              status: 'In Progress',
+            },
+          ],
+        };
+        contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+          contentfulGraphQLResponse,
+        );
+        const response = await workingGroupDataProvider.fetchById(id);
+
+        expect(response?.deliverables).toEqual([
```

Comment:
not a huge deal really but im of an opinion that when dealing with nullable types for properties and variables in tests it is better to assert non-null with `!` rather than do `?` for chaining - reason being: we know we should be getting a response here and if you do `?` this might accidentally default to something you didn't anticipate. Like if you wanted a property to be undefined on an object that should not be undefined then `?` wouldnt really work would it

Reply:
done here [review: replace ? by ! in expected response](https://github.com/yldio/asap-hub/pull/3110/commits/37d06671a698a7d18828c8a943d76861ba931601)